pkg/scanner/reorder.go

```diff
@@ -28,16 +35,51 @@ type ReorderT struct {
 	cb     ScanFuncT
 	window int64
 	clock  int64
+	mUsed  int
+	mLimit int
```

Comment:
nit: I'd call it `memUsed` `memLimit` ... for consistency with `memlimit` that was used below and better naming maybe.